feat: manual alert handling

mwan9ilwa opened the pull request:
Manual Alerts - Feature Implementation

This pull request covers the initial implementation of core functionality for the Manual Alerts module. The following features have been completed:

Completed Tasks:
View Alert Table – Added logic to retrieve and list stored alerts.

Open and Review an Alert – Implemented functionality to fetch and return detailed alert data by ID.

Convert Alert to Case – Added the ability to convert an alert into a case with appropriate data transformation and persistence.

Close Alert as False Positive – Enabled marking and closing of alerts as false positives.

Ingest Alert via Tazama NATS – Integrated with the Tazama NATS broker to consume and process alert messages in real time.

Extract Tenant ID and Transaction Type – Implemented parsing and validation of tenantId and txTp from the transaction object in alert payloads.

Testing and Validation:
Unit and integration tests implemented for core services and processing logic.

Functional validation performed on Manual alert processing workflows.
Tasks
[1. Manual Alerts - View Alert Table]
[2. Manual Alerts - Open and Review an Alert]
[3. Manual Alerts - Convert Alert to Case]
[4. Manual Alerts - Close Alert as False Positive]
[5. Manual Alerts - Ingest Alert via Tazama NATS
[6. Manual Alerts - Extract Tenant ID and Transaction Type from Transaction Object]

mwan9ilwa committed:
feat: implement case management system with React Query

- Add complete frontend application with React, TypeScript, Vite
- Implement React Query for server state management
- Add client-side filtering and sorting for alerts table
- Create triage module for alert management
- Add Prisma schema for database management
- Implement alert operations (update, close, convert to case)
- Add comprehensive component library and utilities

mwan9ilwa committed:
feat: add complete backend infrastructure

- Add authentication and authorization module
- Add audit logging and tracking
- Add case management module
- Add comment system for cases and alerts
- Add task management functionality
- Add NATS messaging integration
- Add alert priority management
- Add Prisma database with migrations
- Add configuration validation
- Add development tooling (husky, prettier, eslint)
- Add comprehensive package management setup

mwan9ilwa committed:
refactor: clean up root directory structure

- Remove duplicate backend files from root directory
- Keep all backend code properly organized in backend/ folder
- Update workspace configuration to remove non-existent shared folder
- Streamline monorepo scripts for frontend and backend only
- Maintain clean separation between frontend/ and backend/ workspaces

mwan9ilwa committed:
feat: add comprehensive test suite and backend tooling

- Add complete unit test coverage for all modules
- Add backend configuration files (.eslintignore, .prettierignore, etc.)
- Add Prettier configuration for consistent code formatting
- Add public key for authentication
- Ensure all .npmrc files use environment variables for security
- Complete test coverage for auth, triage, audit, and other modules

mwan9ilwa committed:
feat: update priority enum, handle null alert types, remove alert status, and add action history endpoint

- Updated Priority enum from LOW/MEDIUM/HIGH/CRITICAL to NEW/URGENT/CRITICAL/BREACH to match backend schema
- Added null handling for alert_type field - frontend now displays 'NULL' when alert_type is explicitly null
- Removed alert_status field and all status-related filtering since alerts don't have status
- Removed status column from AlertsTable and status filters from AlertsSearchAndFilters
- Updated alert transformers to properly handle new priority values and null alert types
- Added missing GET /api/v1/triage/alerts/:alertId/action-history endpoint to backend controller
- Fixed TypeScript typing issue in closeAlert endpoint
- Updated test mocks to reflect new schema changes
- Updated filter options and priority color mappings throughout frontend

UmairKhan-Paysys commented on backend/.env.example:
```diff
@@ -0,0 +1,36 @@
+# =============================================================================
+# INSTRUCTIONS
+# =============================================================================
+# 1. Copy this file to .env: cp .env.template .env
+# 2. Replace all placeholder values with your actual configuration
+# 3. Ensure the public key file exists at the specified path
+# 4. Update DATABASE_URL with your actual database credentials
+# 5. Update TAZAMA_AUTH_URL with your authentication service endpoint
+#
+# SECURITY NOTE: Never commit the .env file to version control!
+# The .env file should be added to .gitignore to prevent accidental commits.
+#Nats Server Configuration
+# =============================================================================
+#Nats Server
```
Prefer we clean up the .env.example file.
```suggestion
```

Akku-Rawat replied:
Thanks for pointing that out, I have cleaned up  .env.example file.

UmairKhan-Paysys commented on backend/.env.example:
```diff
@@ -0,0 +1,36 @@
+
+SYSTEM_UUID=a9a8ff94-c7e4-4e6c-b421-e6d5d75a76e1
+CERT_PATH_PUBLIC=public-key.pem
+CERT_PATH_PRIVATE=private-key.pem
```
We do not need the CERT_PATH_PRIVATE as it is not used. Will need to remove its validation from the `env.validation.ts` as well.
```suggestion
```

Akku-Rawat replied:
have removed and updated env.validation.ts

UmairKhan-Paysys commented on backend/.env.example:
```diff
@@ -0,0 +1,36 @@
+# NATS Server
+STARTUP_TYPE=nats
+NODE_ENV=dev
+DESTINATION_TRANSPORT_URL=nats://10.10.80.30:4222
```
This is not used and not needed. We have fixed out implementation of NATS.
Only  `SERVER_URL` is used for the url of NATS server. So we can remove this.
```suggestion
```

Akku-Rawat replied:
have removed and updated env.validation.ts

UmairKhan-Paysys commented on backend/.env.example:
```diff
@@ -0,0 +1,36 @@
+PRODUCER_STREAM=default
+CONSUMER_STREAM=cms
+OUTPUT_TO_JSON=true
+# Relay service configuration we are using this for Loggerservice
```
Unused and not needed. Kindly remove.
```suggestion
```

Akku-Rawat replied:
have removed

UmairKhan-Paysys commented on backend/package.json:
```diff
@@ -0,0 +1,94 @@
+  "description": "",
+  "author": "",
+  "private": true,
+  "license": "UNLICENSED",
```
```suggestion
  "license": "Apache-2.0",
```

Akku-Rawat replied:
updated license

Akku-Rawat committed:
chore(env): update .env.example and validation schema for new variables

- Added new env variables: AI_TRIAGE_ENABLED, CONFIDENCE_THRESHOLD, CLIENT_SYSTEM_INTERDICTION_ENABLED

- Removed redundant variables: CERT_PATH_PRIVATE, DESTINATION_TRANSPORT_URL

- Updated environment validation schema to include new fields

- Updated package.json license to Apache-2.0

mwan9ilwa committed:
fix: Replace mock user data with auth context and remove alert_status references

- Replace mock user in Layout component with actual auth context
- Replace mock user in NavigationContext with auth context
- Remove alert_status references from VirtualizedAlertTable and AlertDetails
- Update priority values to match new enum (NEW/URGENT/CRITICAL/BREACH)
- Replace alert_status display with alert_type in components
- Remove status filter from useAlerts hook (replaced with type filter)
- Clean up debug files (authServiceDebug.txt, LoginDebug.txt)
- Fix type compatibility issues between auth context and components

mataa019 committed:
feat: fix all the backend test cases

mwan9ilwa committed:
feat: Update priority enum values and make role hierarchy configurable

- Update useAlertOperations.ts to map new priority values (new/urgent/critical/breach) to backend enum
- Fix UpdateAlertModal to use new priority options with proper color mapping
- Update ConvertToCaseData interface to accept new priority values
- Create roles.config.ts for centralized role and permission management
- Move hardcoded ROLE_HIERARCHY to configurable system with proper types
- Add helper functions for role/permission checking (hasPermission, hasAnyRole, getEffectiveRoles)
- Update test files to use new priority values (URGENT instead of MEDIUM)
- Add comprehensive permission mapping for each role level

mwan9ilwa committed:
Merge branch 'manual-alert-handling' of https://github.com/tazama-lf/case-management-system into manual-alert-handling

UmairKhan-Paysys commented on backend/src/alert-priority/alert-priority.service.ts:
```diff
@@ -0,0 +1,92 @@
+@Injectable()
+export class AlertPriorityService implements OnModuleInit {
+  private readonly logger = new Logger(AlertPriorityService.name);
+  private intervalRef: NodeJS.Timeout | null = null;
+  private urgencyThresholds: number[] = [0.33, 0.66, 1.0];
+  private defaultSlaHours: number = 72;
+  private updateIntervalMs: number = 3600000; // 1 hour in milliseconds
```
I believe all of these variables can be configurable. Set them up in .env at the time of deployment and validate them in the .env.validation.ts. This would allow them to be configurable at deployment and not be hard-coded.

UmairKhan-Paysys commented on backend/src/alert-priority/alert-priority.task.ts:
```diff
@@ -0,0 +1,20 @@
+
+  constructor(private readonly priorityService: AlertPriorityService) {}
+
+  @Cron(CronExpression.EVERY_HOUR)
```
We should make it configurable please.

mwan9ilwa committed:
feat: migrate frontend to feature-based architecture

- Organized frontend code into feature modules: alerts, auth, cases, dashboard, shared
- Created feature-specific directories with components, hooks, services, types, utils
- Updated import paths to support new modular structure
- Added index.ts files for clean feature exports
- Moved shared components to shared/components/layout and shared/components/ui
- Relocated services to respective feature modules
- Updated all internal imports to use new paths
- Maintained existing functionality while improving code organization

Features:
- /features/alerts/: Alert management components, hooks, services
- /features/auth/: Authentication components, context, services
- /features/cases/: Case management functionality
- /features/dashboard/: Admin and supervisor dashboards
- /features/shared/: Common UI components, layout, hooks, services

This modular structure improves maintainability, testing, and development workflow.

mwan9ilwa committed:
feat: complete frontend cleanup and finalize feature-based structure

- Moved NotificationProvider and QueryProvider to shared/providers
- Relocated NavigationContext to shared/contexts
- Moved roles.config to shared/config
- Updated all import paths to use new feature-based locations
- Removed unused legacy directories: components/, services/, types/, utils/, styles/
- Cleaned up empty directories and obsolete index files
- Updated App.tsx imports to use feature-based paths

Cleanup Summary:
✅ All providers consolidated in shared/providers
✅ All contexts moved to appropriate feature locations
✅ Configuration files organized in shared/config
✅ Legacy folder structure completely removed
✅ Import paths updated throughout codebase

The frontend now has a clean, organized feature-based architecture with no legacy artifacts.

UmairKhan-Paysys commented on backend/src/alert-priority/alert-priority.service.ts:
```diff
@@ -0,0 +1,92 @@
+          urgency = 'New';
+        }
+
+        const priorityScore = 0.5; // Default priority score
+
+        const updatedData = { ...alertData };
+        updatedData.priority_score = priorityScore;
+        updatedData.urgency = urgency;
+        updatedData.sla_hours = slaHours;
+        updatedData.sla_progress = Number(slaProgress.toFixed(4));
+        updatedData.last_priority_update = now.toISOString();
+
+        await this.prisma.alert.update({
+          where: { alert_id: alert.alert_id },
+          data: {
+            alert_data: updatedData,
+          },
+        });
```
This is the model of Alert we have been following:
```typescript
model Alert {
  alert_id           String             @id @default(uuid()) @db.Uuid
  tenant_id          String             @db.Uuid
  priority           Priority?          @default(NEW)
  alert_type         AlertType?
  prediction_outcome PredictionOutcome? @default(FALSE_POSITIVE)
  source             String?            @db.VarChar(50)
  txtp               String?            @db.VarChar(50)
  message            String             @db.VarChar(100)
  alert_data         Json               @db.JsonB
  transaction        Json               @db.JsonB
  network_map        Json               @db.JsonB
  confidence_per     Int
  created_at         DateTime           @default(now()) @db.Timestamp(6)
  case_id            String             @unique @db.Uuid
  case               Case               @relation(fields: [case_id], references: [case_id])

  @@map("alerts")
}
```

Based on the user story 1-J and the docs. It would be best to add a priority_score in the Alert Model and then to expect the priority_score from the AI-model.
That would mean the `priority` which we are referring to `urgency` here would be calculated in the handler based on the priority_score in thirds

UmairKhan-Paysys replied:
The priority_score can be mapped in thirds like:
<img width="854" height="178" alt="image" src="https://github.com/user-attachments/assets/a7e9b2cb-53e3-4a43-87a4-b150585db686" />

Let's say if the PRIORITY_FIRST_HALF=.33
PRIORITY_SECOND_HALF=.66
PRIORITY_THIRD_HALF=1
are configured.
Based on these threshold you can determing the priority as seen in the diagram.

UmairKhan-Paysys replied:
Also note any update to the alert should reflect an update on the case as well. `priority` specifically.

UmairKhan-Paysys commented on backend/src/auth/auth.decorator.ts:
```diff
@@ -0,0 +1,43 @@
+ * Common Tazama claims for convenience
+ */
+export const TazamaClaims = {
+  CMS_TEST_ROLE: 'CMS-TEST-ROLE',
```
We can possibly rename this and the decorator to something like 'ALERT_TRIAGE' instead of CMS_TEST_ROLE. This will also require an update on keycloak.

mwan9ilwa committed:
feat: Complete frontend reorganization and UI component library implementation

Major Changes:
- Restructured shared folder outside features directory for better architecture
- Created separate admin/ and supervisor/ feature folders
- Removed legacy dashboard folder and DashboardHeader component
- Implemented comprehensive UI component library (Notifications, Containers, Spinners, LoadingState)
- Enhanced authentication system with CMS-TEST-ROLE auto-generation
- Updated all import paths after shared folder relocation

UI Components:
- Notification: 4 types (success, error, warning, info) with toast container
- Container: PageContainer, Card, Section with configurable styling
- Spinner: Multiple variants (spinner, dots, pulse) with text support
- LoadingState: Comprehensive loading states with skeleton components
- AccessibilityComponents: Focus trap, skip-to-content, screen reader support

Code Organization:
- Moved shared/ from features/shared/ to src/shared/
- Created features/admin/ and features/supervisor/ directories
- Updated router to use new dashboard locations
- Fixed all import references across the codebase

Authentication Enhancements:
- Added extractBackendClaims method for improved role extraction
- Auto-generates CMS-TEST-ROLE for backend compatibility
- Enhanced validateBackendAccess function

Build Status:
✅ Frontend builds successfully
✅ All imports resolve correctly
✅ Authentication system enhanced
✅ Clean feature-based architecture maintained

UmairKhan-Paysys commented on backend/src/nats/dto/AlertMessageDto.dto.ts:
This is one of the points where we will need to add `priority_score` to get that 1-J user story right. Along with the changes in the `triage.service.ts` and `triage.controller.ts`

`priority` would be something calculated based on the `priority_score`

UmairKhan-Paysys commented on backend/src/nats/nats.service.ts:
```diff
@@ -0,0 +1,109 @@
+        networkMap: req.networkMap,
+      };
+
+      const alert = await this.triageService.handleNewAlert(
+        submitAlertDto,
+        systemId,
+        tenantId,
+        'NATS',
+      );
```
I see a possible problem. We have three cases when it comes to handling triage.

1. AI Triage is enabled.
2. AI Triage is disabled.
3. Manual Triage.

The different between the 3 is that: 
When AI Triage is enabled a case is created, a 'Triage Task' is created, alert is saved in the db, triage is performed and the alert and case are updated, then the 'Triage Task' is closed (it was assigned to the system). Finally, a task 'Investigate Case' is created which remains UNASSIGNED. Case is also set to a status of `READY_FOR_ASSIGNMENT`

When AI Triage is disabled a case is created, alert is saved in the db and an 'Investigate case' task is directly created (no triage is performed). The task remains UNASSIGNED.

When Manual Triage is enabled, it follows almost the same flow as when AI triage is enabled with an exception that the Triage process is manual and done by a user. The end result remains the same throughout i.e. an 'Investigate case' task is created, which remains UNASSIGNED an up for taking. 

I believe we are not handling these three scenarios.

Akku-Rawat replied:
The first two scenarios were already being handled, and we have now added manual triage handling

UmairKhan-Paysys commented on backend/src/nats/nats.service.ts:
```diff
@@ -0,0 +1,109 @@
+          {
+            caseId: alert.case_id,
+            assignedUserId: systemId,
+            status: TaskStatus.ASSIGNED_10,
```
Should be UNASSIGNED. we do not handle automatic assignments yet. As we have no way to decide to whom the task should be assigned.

UmairKhan-Paysys commented on backend/src/triage/dto/convert-alert-to-case.dto.ts:
Not needed anymore

UmairKhan-Paysys commented on backend/src/triage/triage.service.ts:
```diff
@@ -0,0 +1,704 @@
+    private configService: ConfigService,
+  ) {}
+
+  async handleNewAlert(alert: SubmitAlertDto, userId: string, tenantId: string, source: string) {
```
This handles the AI Triage is disabled part. Not the manual part.

Akku-Rawat replied:
Have added manual handling

UmairKhan-Paysys commented on backend/src/triage/triage.service.ts:
```diff
@@ -0,0 +1,704 @@
+      },
+    });
+
+    if (!alert) {
+      throw new NotFoundException(`Alert with ID ${alertId} was not found for tenant ${tenantId}.`);
+    }
+
+    try {
+      const existingCase = await this.caseService.retrieveCase(alert?.case_id);
+
+      if (
+        existingCase.status === CaseStatus.CLOSED_CONFIRMED_82 ||
+        existingCase.status === CaseStatus.CLOSED_REFUTED_81 ||
+        existingCase.status === CaseStatus.CLOSED_INCONCLUSIVE_83
+      ) {
+        throw new BadRequestException(`Case ${existingCase.case_id} linked with alert ${alertId} is already closed`);
+      }
+
+      const closedCase = await this.caseService.updateCase(existingCase.case_id, { status: closeAlertDto.status }, userId);
+
+      const createCommentDto = new CreateCommentDto();
+      createCommentDto.caseId = closedCase.case_id;
+      createCommentDto.note = closeAlertDto.reason;
+
+      this.commentService.addComment(createCommentDto, userId);
+
+      await this.audit.logAction({
+        userId,
+        operation: 'ALERT_CLOSED',
+        entityName: 'Alert',
+        actionPerformed: `Closed case for alert ${alertId} with reason: ${closeAlertDto.reason}  at ${new Date().toISOString()}`,
+        outcome: Outcome.SUCCESS,
+      });
+
+      return closedCase;
+    } catch (error) {
+      this.logger.error(`Failed to close case for alert ${alertId} : ${error.message}`, TriageService.name);
+      throw new InternalServerErrorException('Failed to close alert');
+    }
+  }
```
UpdateAlert and ManualClose can be turned to a single function. I.e. for manual triage. Keep in mind alert update should only be available if `TRIAGE_TYPE` is configured to manual.

Akku-Rawat replied:
Created handleManualtriage for this, also updated endpoint and request payload

UmairKhan-Paysys commented on frontend/README.md:
Would prefer better detail regarding the frontend

UmairKhan-Paysys commented on frontend/public/vite.svg:
Would be cool to have Tazama logo display as the tab icon instead of Vite logo.

mwan9ilwa replied:
Done

UmairKhan-Paysys commented on frontend/src/components/accessibility/AccessibilityComponents.tsx:
Love the effort to ensure accessibility. Appreciate it!

UmairKhan-Paysys commented on frontend/src/components/auth/ProtectedRoute.tsx:
```diff
@@ -0,0 +1,62 @@
+      (role) => user.roles.includes(role) || user.roles.includes('admin'), // Admin has access to everything
+    );
+
+    if (!hasRequiredRole) {
+      return (
+        <div className="min-h-screen flex items-center justify-center">
+          <div className="text-center">
+            <h2 className="text-2xl font-bold text-gray-900 mb-2">
+              Access Denied
+            </h2>
+            <p className="text-gray-600 mb-4">
+              You don't have permission to access this page.
+            </p>
+            <button
+              onClick={() => window.history.back()}
+              className="btn btn-primary"
+            >
+              Go Back
+            </button>
+          </div>
+        </div>
+      );
+    }
+  }
```
This is great but as a suggestion it would be better to not be able to see any option a user does not have access to.

mwan9ilwa replied:
the frontend now as of this afternoon, properly handle the CMS-TEST-ROLE claim that the backend expects.

mwan9ilwa replied:
The backend has Role-Based Access Control, but it's currently limited

Current RBAC Implementation:
1. Claims-Based Authentication:
    * Uses @tazama-lf/auth-lib for token validation
    * Implements claims-based access control via JWT tokens
    * Has a robust [TazamaAuthGuard] that validates tokens and claims
2. Authorization Decorators:
    * @RequireClaim(claim) - Single claim requirement
    * @RequireClaims(...claims) - Multiple claims requirement (all must be present)
    * @Public() - Marks routes as public (no auth required) 
3. Current Protection:
    * All endpoints currently use @RequireCMSTestRole()
    * All controllers (triage, case, comment, task) are protected
    * Authentication guard extracts user info from JWT tokens

Current Limitations:
1. Single Role Implementation:
    * Currently only uses CMS-TEST-ROLE for all endpoints
    * No granular role separation (admin, supervisor, analyst, investigator)
    * All authenticated users have the same access level
2. Missing Role Hierarchy:
    * No distinction between different user roles
    * No endpoint-specific permissions based on user roles
    * No resource-level access controls

UmairKhan-Paysys commented on frontend/src/hooks/useAsyncOperation.ts:
Suggest removing empty files. Thank you

UmairKhan-Paysys reviewed (changes requested):
Left comments for the backend.

As for the frontend I thought it would be better to give is here in a single comment.
- We no longer convert to case so that button needs to be updated to and to make sure it is only visible if Manual Triage is configured. I.e. its a manual triage button and not a convert to case or update alert button.
- close alert should be part of manual triage and not a separate operation anymore.
- I understand the backend is constantly updating so there are api calls which require updates.
- Kindly make it possible to hide the features if the user does not have access to of if AI Triage is enabled then there should be no option to update alert available.

I recommend coordinating with the backend team for updating. Thanks

mwan9ilwa committed:
Added the tazama logo and reduced table and search bar shadow

Akku-Rawat committed:
refactor(ai-triage): updated investigate case task status to unassigned

Akku-Rawat committed:
fix: test cases

dev-Gerald committed:
fixed alert priority

dev-Gerald committed:
alert priority

Akku-Rawat committed:
feat(triage): add manual triage

mwan9ilwa committed:
fixed pagination

mwan9ilwa committed:
Merge branch 'manual-alert-handling' of https://github.com/tazama-lf/case-management-system into temp/pr-feedback-fixes

mwan9ilwa committed:
feat: streamline Alert module with comprehensive Manual Triage workflow

- Remove redundant UpdateAlertModal and CloseAlertModal components
- Add ManualTriageModal for unified alert triage operations
- Update AlertsDetailModal to use single Manual Triage action button
- Enhance TriageService with performManualTriage method
- Add ManualTriageDto type and update related interfaces
- Update AlertsDashboard to handle manual triage workflow
- Clean up component exports and dependencies

This consolidates all alert update and case decision operations into a single,
comprehensive manual triage interface that aligns with the sophisticated backend
triage system.

Akku-Rawat committed:
fix(manual-triage): only allow assigned user to do manual triage

Akku-Rawat committed:
refactor: update getAlertsForUser to return transaction

mwan9ilwa committed:
getting new changes

mwan9ilwa committed:
Merge branch 'manual-alert-handling' of https://github.com/tazama-lf/case-management-system into temp/pr-feedback-fixes

Akku-Rawat committed:
refactor(manual-triage): update manual triage impl for priority and alertType updates

added priorityScore to manual triage request body

added NONE to alertType enum

Akku-Rawat committed:
refactor: update predictAlert to return priorityScore

mwan9ilwa committed:
feat: implement 3-mode triage system UI integration

- Add backend system configuration endpoint (/api/v1/config/system)
- Create SystemConfigService and useSystemConfig hook for frontend
- Add conditional UI behavior based on triage mode:
  * AI Mode: Show 'AI Processed' indicators, hide manual controls
  * Manual Mode: Full triage form with prediction outcome and case status
  * Disabled Mode: Simplified form, direct investigation routing
- Update AlertsDetailModal with mode-specific buttons and indicators
- Update ManualTriageModal with conditional fields based on mode
- Add dynamic subtitles based on current triage configuration
- Create TriageModeIndicator component (available but not displayed)
- System automatically adapts UI based on TRIAGE_TYPE environment variable

Supports seamless switching between AI, MANUAL, and DISABLED triage modes
without code changes - only environment configuration updates needed.

mwan9ilwa committed:
feat: Add priority score functionality and improve UI

- Add priorityScore field to backend ManualTriageDto with validation
- Implement automatic priority calculation based on configurable thresholds
- Update frontend to support priorityScore in manual triage modal
- Add priority score input field with proper validation (0-1 range)
- Update backend predictAlert method to include priorityScore
- Add NONE enum values to CaseType and AlertType
- Truncate Alert ID and Transaction ID columns to 8 characters in table
- Improve manual triage form with better AlertType options
- Ensure frontend/backend API compatibility for priority scoring

mataa019 committed:
fix all backend test

mataa019 committed:
Merge branch 'manual-alert-handling' of https://github.com/tazama-lf/case-management-system into manual-alert-handling

mataa019 committed:
fix all backend test

mwan9ilwa committed:
fix: resolve database schema sync and refactor transaction messages

- Add missing priority_score column to alerts table migration
- Fix AlertType enum alignment between frontend and backend (FRAUD, AML, FRAUD_AND_AML)
- Refactor TransactionMessagesModal to extract messages from alert data instead of separate API calls
- Add transaction utility functions for extracting messages and IDs from alert data
- Add toast notifications and error handling for manual triage operations
- Add loading skeletons and improved UX components
- Update manual triage DTO to include priorityScore field
- Synchronize database schema with Prisma model definitions

mwan9ilwa committed:
pulling latest changes

Akku-Rawat committed:
chore: update env.example file

- removed ununsed close alert dto

- removed close alert dto test cases

dev-Gerald committed:
case dashboard

dev-Gerald committed:
case dashboard

dev-Gerald committed:
alert priority fixed

mwan9ilwa committed:
Add/update Dockerfile for multi-stage build

mwan9ilwa committed:
Merge branch 'manual-alert-handling' of https://github.com/tazama-lf/case-management-system into manual-alert-handling

mwan9ilwa committed:
fix: resolve frontend bugs in triage system

- Fix TRIAGE_TYPE=DISABLED showing Update Alert button incorrectly
  * Updated button visibility condition to exclude DISABLED mode
  * Added missing PencilSquareIcon import
  * Fixed button click handler to use proper onManualTriage callback

- Fix validation errors not clearing in ManualTriageModal
  * Added real-time validation that clears errors when form values change
  * Enhanced form reset to include modal open/close state
  * Clear validation errors on successful form submission

- Fix case status not updating in real-time in Alert Details
  * Enhanced alert operation mutations to invalidate case queries
  * Added robust case ID resolution with fallback to cached data
  * Ensures case status updates immediately without page refresh

Resolves frontend functionality issues with triage mode handling,
form validation UX, and real-time data synchronization.

Akku-Rawat committed:
fix(triage): fixed triage issue as per the feedback

Prevent updating a triage task once marked Completed

Preserve generic task description ('Task for Manual Triage') and add comments separately

Ensure case details syncs with alert details after manual triage

Prevent duplicate comments and multiple 'Investigate Case' tasks for the same case

dev-Gerald committed:
Triage user assignment & test cases

dev-Gerald committed:
fixed work flow for task assignment

Akku-Rawat committed:
fix(manual-triage): resolve auto-assignment and add comments for triage

- Prevent automatic assignment of investigation tasks

- Add comment for triage task instead of updating description

- Use generic triage description ('Task for Manual Triage')

- Fix failing test cases

dev-Gerald committed:
updated readme

dev-Gerald committed:
updated readme

UmairKhan-Paysys reviewed (changes requested):
# ATM TestingNFeedback

## Cases:

### AI Enabled

1. 1.B High confidence and FALSE_POSITIVE => AUTOCLOSED_REFUTED											(Working)
2. 1.C High confidence and TRUE_POSITIVE => AUTOCLOSED_CONFIRMED										(Needs Testing)
3. 1.D High confidence, TRUE_POSITIVE, and transaction occured => Move to investigations				(FIXED)
4. 1.E Money Laundering => Move to investigations														{Working)
5. 1.F Low confidence, TRUE_POSITIVE, AML/FRAUD, Transaction not BLOCKED => Move to investigations		(FIXED)
6. 1.G No confidence threshold defaults to => Move to investigationsi									(FIXED)
7. 1.I Split cases in case both Fraud and AML is discovered => Split and move to investigations			(FIXED)

### Triage is Disabled

1. 1.H If Triage is DISABLED => Move to investigations with no Triage Task								(Default ALRT predition should be TRUE_POSITIVE)

### Manual Triage

1. User should be able to perform manual triage.														(Needs a split implementation)
	-	CLOSED_CONFIRMED
	-	CLOSED_REFUTED
	-	MOVE TO INVESTIGATION
	
### SLA

1. 1.J SLA to update priority based on priority_score.

_note:_ Considering confidence threshold as 95

## Observations:

### AI Enabled

1. AI_TRIAGE_ENABLED env is not used and should be removed
2. Priority does not update in case of AI Triage
3. In case of AI_Triage the alert prediction_outcome is not updated
4. ALRT priority is not reflected in the Case priority
5. Little typo in case of AML prediction. Task is created with title of 'FRAUD'
6. In case of low confidence percentage the alert type is not reflected in the case
7. For no confidence threshold config the CONFIDENCE_THRESHOLD needs to be optional in config validation
8. In case of both FRAUD_AND_AML the priority is not being passed which results in error
9. In case of both FRAUD_AND_AML the parentId of the case is not being set.
10. In case of both FRAUD_AND_AML the parent case is not updated.
11. Identified an issue that if Triage is Disabled the default prediction_outcome should be TRUE_POSITIVE (schema.prisma needs and update)

### Disabled

1. The default alert prediction_outcome should be set to TRUE_POSITIVE

### Manual

1. In case of FRAUD_AND_AML cases do not split.

### UI

1. Order on risk score breaks
3. Only Source and Priority filters work. Rest of them do nothing or filter out all alerts.
4. The scroll on Show Risk Breakdown does not work. It is cut off
5. Risk Type should be a description in the Risk Breakdown

### General Update

1. Case_Status should start with a Prefix
__Example__
UNASSIGNED_01 => STATUS_01_UNASSIGNED
2. Comments should be related to tasks instead of cases.

### Docs 

1. Should mention the DB setup (npx prisma generate; npx prisma migrate dev)
2. README.md does not cover the deployment setup completely

### Most fixes done in the fork for Demo

Please refer [here](https://github.com/UmairKhan-Paysys/case-management-system/tree/manual-alert-handling)

Akku-Rawat committed:
fix(alert-triage): fix bugs encountered

- Fixed priority not being updated in cases associated with alerts

- Made CONFIDENCE_THRESHOLD optional in config

- Propagated low-confidence alert type to cases

- Set default prediction_outcome to TRUE_POSITIVE

- Synced details between parent and child cases when alert type is FRAUD_AND_AML

- Created type for AI Prediction response

Akku-Rawat committed:
fix(manual-triage): handle FRAUD_AND_AML alert type correctly

- Fixed case not splitting when alert type is FRAUD_AND_AML

- Updated comments to be associated with task instead of case

Akku-Rawat committed:
refactor(alert-triage): Update CaseStatus and TaskStatus enums to STATUS_XX_DESCRIPTION format

Akku-Rawat committed:
refactor(alert-triage-ui): Update CaseStatus and TaskStatus enums to STATUS_XX_DESCRIPTION format

mwan9ilwa committed:
fix: enable source and time range filters by switching to client-side filtering

- Switch AlertsDashboard to use client-side useAlerts hook instead of server-side useAlertsQuery
- Enables time range filtering (today, yesterday, this week, custom date ranges) which was not supported by backend
- Enables proper source filtering with client-side processing
- Maintains all existing functionality while adding working filter capabilities
- Uses local state management for comprehensive filtering and sorting

mwan9ilwa committed:
feat: improve Rules & Typologies table UX in AlertsDetailModal

- Display subRuleRef in Type column for rule results instead of generic type
- Add scrollable table with sticky header and footer for better navigation
- Enable vertical scrolling when many rule results are present
- Keep column headers visible during scrolling for better user experience
- Keep total score row pinned at bottom for constant visibility
- Set max height of 320px for scrollable content area

mwan9ilwa committed:
docs: comprehensive README updates for better developer onboarding

Root README:
- Add quick start guide with prerequisites and setup steps
- Include comprehensive project structure and architecture overview
- Document monorepo commands and individual service commands
- Add detailed environment configuration examples
- Include deployment, security, and technology stack information
- Provide contributing guidelines and API documentation

Frontend README:
- Complete rewrite with modern React development practices
- Add detailed architecture and tech stack information
- Include comprehensive testing strategy and commands
- Document code style, performance optimization, and accessibility
- Add deployment guidelines and development workflow
- Provide extensive resource links and development tools

These updates will help new developers understand the project structure,
setup requirements, and development practices more effectively.

mwan9ilwa committed:
feat: display userId in action history

- Add userId display in action history for both AlertsDetailModal and AlertDetails components
- Show userId next to timestamp with bullet separator for better readability
- Conditionally render userId only when available
- Maintain existing styling and layout structure
- Improves traceability by showing who performed each action

UmairKhan-Paysys reviewed (dismissed):
LGTM.

blessedjasonmwanza committed:
Add Workflow to publish docker image
